dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TaskDefinitionServiceImpl.java

```diff
@@ -846,10 +852,15 @@ public Map<String, Object> updateTaskWithUpstream(User loginUser, long projectCo
         } else {
             queryUpStreamTaskCodeMap = new HashMap<>();
         }
-        if (CollectionUtils.isNotEmpty(upstreamTaskCodes)) {
+        if (queryUpStreamTaskCodeMap.size() != 0) {
             ProcessTaskRelation taskRelation = upstreamTaskRelations.get(0);
             List<ProcessTaskRelation> processTaskRelations =
                     processTaskRelationMapper.queryByProcessCode(projectCode, taskRelation.getProcessDefinitionCode());
+
+            // set upstream code list
+            updateUpstreamTask(new HashSet<>(queryUpStreamTaskCodeMap.keySet()),
```

Comment:
```suggestion
            updateUpstreamTask(queryUpStreamTaskCodeMap.keySet(),
```
You don't need to new set here?